engine.py

```diff
@@ -882,12 +949,136 @@ def _get_dialog_parent(self):
                 # try to get a hold of the main window. it seems to be the only
                 # one with windowIconText set. There should be a better way to
                 # do this.
-                if (widget.isWindow() and
-                        not isinstance(widget, QtGui.QDialog) and
-                        widget.windowIconText()):
+                if (
+                    widget.isWindow()
+                    and not isinstance(widget, QtGui.QDialog)
+                    and widget.windowIconText()
+                ):
                     parent = widget
 
-        self.logger.debug(
-            "Found top level widget %s for dialog parenting" % (parent,))
+        self.logger.debug("Found top level widget %s for dialog parenting" % (parent,))
         return parent
 
+    def node_handler(self, node):
+        """
+        Get the node handler hook for the given node.
+
+        :param node: A :class:`hou.Node` instance.
+
+        :returns: A :class:`NodeHandlerBase` instance.
+        """
+        if not node:
+            return
+        node_type = node.type()
+        node_type_name = node_type.name()
+        node_category = node_type.category().name()
+        hook_instance = self.__node_handlers.get(node_category, {}).get(
+            node_type_name, False
+        )
+
+        if hook_instance is False:
+            if not self.__node_handlers.get(node_category):
+                self.__node_handlers[node_category] = {}
+            node_handlers = self.get_setting("node_handlers")
+            hook_instance = None
+            for handler in node_handlers:
+                setup_hook = (
+                    handler["node_category"] == node_category
+                    and handler["node_type"] == node_type_name
+                )
+                if setup_hook:
+                    tk_houdini = self.import_module("tk_houdini")
+                    base_class = tk_houdini.base_hooks.NodeHandlerBase
+                    hook_instance = self.create_hook_instance(
+                        handler["hook"], base_class=base_class,
+                    )
+                    break
+            self.__node_handlers[node_category][node_type_name] = hook_instance
```

Comment:
If you choose to do the `setdefault` suggestion above

```suggestion
            category_handler[node_type_name] = hook_instance
```